### PR TITLE
Add more configuration to the Max Layout

### PR DESCRIPTION
**Pull request by Borisd93**
Add more configuration's to the Max QTile Layout.

### CODE REVIEW

File: libqtile/layout/max.py [hunk 1]
```diff
@@ -31,6 +31,15 @@ class Max(_SimpleLayoutBase):
     commands to switch to next and previous windows in the stack.
     """
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if hasattr(self, "margin") is False:
```
Comment (elParaguayo) [inline libqtile/layout/max.py:36]: Properties should be set via `defaults` not here.

**Comment by elParaguayo** [comment]
See #2726 - this is a better approach (but doesn't have borders included). You might want to start with that one and include borders too.

File: libqtile/layout/max.py [hunk 2]
```diff
@@ -31,6 +31,16 @@ class Max(_SimpleLayoutBase):
     commands to switch to next and previous windows in the stack.
     """
 
+    __defaults__ = [
```
Comment (elParaguayo) [inline libqtile/layout/max.py:34]: This should be `defaults` not `__defaults__`.

File: libqtile/layout/max.py [hunk 3]
```diff
@@ -31,6 +31,16 @@ class Max(_SimpleLayoutBase):
...
+        ("margin", 0, "Margin of the layout (int or list of ints [N E S W])"),
+        ("border_color", "#0000ff", "Border colour(s) for the window"),
+        ("border_width", 1, "Border width."),
```
Comment (elParaguayo) [inline libqtile/layout/max.py:37]: Given that the code currently has `0` hard coded for the value, we should use that as the default here.

**Comment by elParaguayo** [comment]
We need a test for this i.e. if the margin is set, the window size should reduce.

**Comment by m-col** [comment]
Hola, could you add an entry to the changelog so that other users find your new features?

**Comment by elParaguayo** [comment]
The default `border_width` is still showing as `1`. This needs to be changed to `0`. As it is, this change will add a border for users who just have `layout.Max()` in their configs which should not happen.

**Review by elParaguayo — APPROVED** [review]
Thanks. LGTM.

**Comment by elParaguayo** [comment]
One small change needed. Can you then squash all the commits and force push this branch again.

**Comment by Borisd93** [comment]
> One small change needed. Can you then squash all the commits and force push this branch again.

Clear the commit's?

**Comment by elParaguayo** [comment]
There's still a typo that needs fixing and the commit history looks a bit off. The `SIGUSR1` commit shouldn't be part of this.

**Comment by hamensman** [comment]
If this is of any use, here is a discussion I made here about adding border focus to the max layout: https://github.com/qtile/qtile/discussions/2891

**Review by elParaguayo — APPROVED** [review]
Made some changes and pushed to Borisd93's fork.

**Review by m-col — APPROVED** [review]
LGTM :)

**Pull request merged**